sway-error/src/error.rs

```diff
@@ -416,6 +416,8 @@ pub enum CompileError {
     MoreThanOneEnumInstantiator { span: Span, ty: String },
     #[error("This enum variant represents the unit type, so it should not be instantiated with any value.")]
     UnnecessaryEnumInstantiator { span: Span },
+    #[error("`{ty}` is a unit enum variant, and does not take parentheses to be constructed.")]
```

Comment:
Can we reword this to:

"The enum variant `{ty}` is of type `unit`, so its constructor does not take arguments or parenthesis. Try removing the `()`."

Reply:
Applied the suggested change.